### PR TITLE
Changes made to DeployDataCenter to support Baremetal Basic Zone Creation

### PR DESCRIPTION
**Pull request by sanju1010**
1. Made changes to deployDataCenter.py to support zone creation with Baremetal as the Hyeprvisor
2. Added test file to BVT with basic vm deployment test 
3. Added template entry in test_data.py file
4. Added sample configuration file BareMetal.cfg to create Basic Zone with Baremetal as the Hypervisor

I have also tested this and works fine. However, I don't have test results to post here.

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@sanju1010 I had a look and didn't see you add your test to the travis yaml, is it included in the travis run? (if yes no need for any test report)

File: test/integration/smoke/test_baremetal.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        )
+        #Update host with host tag
+        cls.hosts = Host.list(
+            cls.apiclient,
+            zoneid=cls.zone.id,
+            hypervisor=cls.hypervisor
+        )
+        if not validateList(cls.hosts)[0]:
+            assert False, "Invalid response for list host"
+        host_res = Host.update(
```
Comment (harikrishna-patnala) [inline test/integration/smoke/test_baremetal.py]: @sanju1010 I have checked the code and there seems a bug exists during vm deployment. Here you are setting the tag to Host using updateHostCmd which puts the host tag in "host_tags" table, but Baremetal planner searches for hosts with hosttag in "host_details" table. Can you please confirm or recheck whether baremetal instance provisioning is successful. If not we need to set the host tag while adding the host itself.

File: tools/marvin/marvin/config/BareMetal.cfg [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+            "value": "1000"
+        },
+        {
+            "name": "enable.dynamic.scale.vm",
+            "value": "true"
+        },
+        {
+            "name": "execute.in.sequence.hypervisor.commands",
+            "value": "true"
+        }
```
Comment (harikrishna-patnala) [inline tools/marvin/marvin/config/BareMetal.cfg]: @sanju1010 , we also need to set the configuration parameter "baremetal.provision.done.notification.enabled" to false. This parameter is applicable only for Baremetal advanced zone.

File: test/integration/smoke/test_baremetal.py (hunk 1 above)
Comment (harikrishna-patnala) [inline test/integration/smoke/test_baremetal.py]: I have checked this scenario, vm deployment is getting succeeded but we need to fix the baremetal planner a bit. So here no need to change anything.

**Comment by sanju1010** [comment]
@DaanHoogland , this test requires hardware. So I can't add this as part of travis yaml.

**Comment by DaanHoogland** [comment]
@sanju1010 makes sense and then of course a test report and -description is needed.

**Comment by harikrishna-patnala** [comment]
I have not tested but did a code walk through. LGTM

**Comment by DaanHoogland** [comment]
@sanju1010 can you add a run output to the PR?

**Comment by sanju1010** [comment]
Sure Daan I will add .

**Comment by rhtyd** [comment]
@sanju1010 please rebase against latest master and push -f, update on status of your PR

This is baremetal specific test, how do I test it; the resources/urls mentioned in your cfg are specific to some private network, in which case the PR test-data is unacceptable if no-one else can run your test

**Comment by DaanHoogland** [comment]
@harikrishna-patnala @sanju1010 closing this stale ticket, feel free to rebase and re-open

**Pull request closed without merging**